### PR TITLE
Add optional HMAC slack OAuth flow

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Add optional HMAC on the Slack credentials. 
Add validation on the API request.
### Why was this change needed?

In order to secure the public API endpoint.

### CODE REVIEW

File: apps/api/src/app/subscribers/usecases/chat-oauth/chat-oauth.usecase.ts [hunk 1]
```diff
@@ -38,13 +76,45 @@ export class ChatOauth {
       );
     }
 
+    if (integration.credentials == null) {
```
Comment (ainouzgali) [inline apps/api/src/app/subscribers/usecases/chat-oauth/chat-oauth.usecase.ts:79]:
```suggestion
    if (!integration.credentials) {
```

Is there a reason not to do it this way? Feels cleaner

**Review by ainouzgali — APPROVED** [review]
🚀

File: apps/web/src/pages/integrations/IntegrationsStorePage.tsx [hunk 2]
```diff
@@ -190,6 +190,8 @@ export interface ICredentials {
   requireTls?: boolean;
   ignoreTls?: boolean;
   tlsOptions?: Record<string, unknown>;
+  redirectUrl?: string;
+  hmac?: boolean;
```
[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/web/src/pages/integrations/IntegrationsStorePage.tsx:194]: It should be auto-generated due to the update of the file 
libs/shared/src/consts/providers/credentials/provider-credentials.ts 
in this PR.

[the remark this answers is not in the document]
Comment (ainouzgali) [reply apps/web/src/pages/integrations/IntegrationsStorePage.tsx:194]: To IntegrationsStoreModal.tsx ? - the code there is duplicated

[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/web/src/pages/integrations/IntegrationsStorePage.tsx:194]: Yeah, should be ok.

[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/web/src/pages/integrations/IntegrationsStorePage.tsx:194]: Both of them should consume the same provider list with the credentials configuration.

**Pull request merged**